src/services/creategeojsonbloodhound.js

```diff
@@ -58,7 +61,14 @@ ngeo.createGeoJSONBloodhound = function(options, opt_featureProjection,
       },
       filter: function(parsedResponse) {
         var featureCollection = /** @type {GeoJSONFeatureCollection} */
-            (parsedResponse);
+            (goog.object.clone(parsedResponse));
```

Comment:
When you use a service (one url) multiple times with differents configuration of datasources, Typeahead will do only one request to this service (this is expected). But when you filter the results in one datasource, the results of this filter will be the new "parsedResponse" from each "same-service" datasource.

So I must work on a cloned object to avoid modifications on the service's response.

(Request a service multiple times is used to make groups in Typeahead.)